test/e2e/service.go

```diff
@@ -797,6 +797,75 @@ var _ = Describe("Services", func() {
 	})
 })
 
+var _ = Describe("Services", func() {
+	f := NewFramework("services-temp")
+
+	var t *explorerTest
+	BeforeEach(func() {
+		t = newDefaultExplorerTest(f.Client, f.Namespace.Name)
+	})
+
+	It("should serve round-robin services round-robin", func() {
+		_, _, svc := t.setup()
+
+		nodeName := getFirstNodeNameOrDie(f.Client)
+
+		h := histogram{}
+		var lastHostname string
+		sampleSize := 20
+		for i := 0; i < sampleSize; i++ {
+			// we have to ssh in order to view the cluster through the lense of a Pod,
+			// i.e. this is the easiest way to only proxy through a single kube-proxy.
+			// unfortunately we can't use the apiserver proxy for this.
+			stdout, stderr, code, err := SSH("curl --silent http://"+svc.Spec.ClusterIP+"/hostname/", nodeName, testContext.Provider)
+			Expect(err).NotTo(HaveOccurred(), "error executing ssh: %v", err)
+			Expect(code == 0).To(BeTrue(), "error code was expected to be 0, but was %d: %s", code, stderr)
```

Comment:
Good to mention where the error code came from here, otherwise the logs look rather cryptic ("expected 0 but got 1" etc).